codex/indexingstrategy.nim

```diff
@@ -39,22 +39,11 @@ func checkIteration(self: IndexingStrategy, iteration: int): void {.raises: [Ind
       IndexingError,
       "Indexing iteration can't be greater than or equal to iterations.")
 
-proc getIter(first, last, step: int): Iter[int] =
-  var
-    finish = false
-    cur = first
+proc getIterSE(first, last, step: int): Iter[int] =
+  newIter(first, last, step)
 
-  func get(): int =
-    result = cur
-    cur += step
-
-    if cur > last:
-      finish = true
-
-  func isFinished(): bool =
-    finish
-
-  Iter.new(get, isFinished)
+func getIter(first, last, step: int): Iter[int] =
+  (cast[proc (f, l, s: int): Iter[int] {.nimcall, noSideEffect.}](getIterSE)(first, last, step))
```

Comment:
`cast` is totally unsafe and its bypassing all the language safety checks, I wonder why we're using it here instead of a safer type conversion?

Reply:
This is inside of a `func` so it can't have side effects. The call to newAsyncIter is a `proc` and can't be turned into a `func`. Is there any other way to solve this mismatch other than casting?

Reply:
By your suggestion I used `{.cast(noSideEffect).}` to do a bit safer casting than previously.